rmgpy/data/kinetics/groups.py

```diff
@@ -119,39 +119,63 @@ def getReactionTemplate(self, reaction):
 
         # Descend reactant trees as far as possible
         template = []
-        for entry in forwardTemplate:
-            # entry is a top-level node that should be matched
+        specialCases = ['peroxyl_disproportionation','bimolec_hydroperoxide_decomposition','r_recombination']
+        if len(forwardTemplate) == 1 and len(reaction.reactants) > len(forwardTemplate) and self.label.lower().split('/')[0] not in specialCases:
+            entry = forwardTemplate[0]
```

Comment:
Could you add a comment about how this if case is for automatically generated trees with only one tree? It took me a while to figure that out... (unless that's wrong, in which case you should definitely add a comment).

Reply:
added in atg4